torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -771,6 +800,98 @@ struct GraphFuser {
     }
   }
 
+  bool usedOnlyInSize(Value * v) {
+    const auto & uses = v->uses();
+    return std::all_of(uses.begin(), uses.end(),
+                       [](const Use& u) { return u.user->matches("aten::size(Tensor self) -> int[]"); });
+  }
+
+  std::unordered_map<Value*, Value*> buildShapeExpressions(Node * fusion_group) {
```

Comment:
Are we building shape expressions for (all values) / (all outputs) / (something else) ?

Reply:
All values (including outputs). Some (even most) of them might end up being unused, and will be cleaned up by DCE. Will try to add a comment (possibly in a next PR).

Reply:
Changing the name to buildShapeExpressionsForAllValues or something would be nice, but yeah, can be done in a future pr